recipes/qt/5.x.x/conanfile.py

```diff
@@ -0,0 +1,710 @@
+
+    def export(self):
+        self.copy("qtmodules%s.conf" % self.version)
+
+    def build_requirements(self):
+        if tools.os_info.is_windows and self.settings.compiler == "Visual Studio":
+            self.build_requires("jom/1.1.3")
+        if self.options.qtwebengine:
+            self.build_requires("ninja/1.10.2")
+            # gperf, bison, flex, python >= 2.7.5 & < 3
+            if self.settings.os != "Windows":
+                self.build_requires("bison/3.7.1")
+                self.build_requires("gperf/3.1")
+                self.build_requires("flex/2.6.4")
+
+            # Check if a valid python2 is available in PATH or it will failflex
+            # Start by checking if python2 can be found
+            python_exe = tools.which("python2")
+            if not python_exe:
+                # Fall back on regular python
+                python_exe = tools.which("python")
+
+            if not python_exe:
+                msg = ("Python2 must be available in PATH "
+                       "in order to build Qt WebEngine")
+                raise ConanInvalidConfiguration(msg)
+            # In any case, check its actual version for compatibility
+            from six import StringIO  # Python 2 and 3 compatible
+            mybuf = StringIO()
+            cmd_v = "{} --version".format(python_exe)
+            self.run(cmd_v, output=mybuf)
+            verstr = mybuf.getvalue().strip().split('Python ')[1]
+            if verstr.endswith('+'):
+                verstr = verstr[:-1]
+            version = tools.Version(verstr)
+            # >= 2.7.5 & < 3
+            v_min = "2.7.5"
+            v_max = "3.0.0"
+            if (version >= v_min) and (version < v_max):
+                msg = ("Found valid Python 2 required for QtWebengine:"
+                       " version={}, path={}".format(mybuf.getvalue(), python_exe))
+                self.output.success(msg)
+            else:
+                msg = ("Found Python 2 in path, but with invalid version {}"
+                       " (QtWebEngine requires >= {} & < "
+                       "{})".format(verstr, v_min, v_max))
+                raise ConanInvalidConfiguration(msg)
+
+    def config_options(self):
+        if self.settings.os != "Linux":
+            del self.options.with_icu
+            del self.options.with_fontconfig
+            del self.options.with_libalsa
+        if self.settings.compiler == "apple-clang":
+            if tools.Version(self.settings.compiler.version) < "10.0":
+                raise ConanInvalidConfiguration("Old versions of apple sdk are not supported by Qt (QTBUG-76777)")
+        if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "5.3":
+            del self.options.with_mysql
+        if self.settings.os == "Windows":
+            self.options.with_mysql = False
+            self.options.opengl = "dynamic"
+        if self.settings.os == "Macos":
+            del self.settings.os.version
```

Comment:
🤔 I've never done native application on MacOS but I'd of assumed this was relevant